independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanResolver.java

```diff
@@ -45,14 +43,20 @@ public BeanResolver(BeanDeployment beanDeployment) {
         this.assignableFromMap = new ConcurrentHashMap<>();
         this.assignableFromMapFunction = name -> {
             Set<DotName> assignables = new HashSet<>();
-            Collection<ClassInfo> subclasses = beanDeployment.getIndex().getAllKnownSubclasses(name);
-            for (ClassInfo subclass : subclasses) {
+            for (ClassInfo subclass : beanDeployment.getBeanArchiveIndex().getAllKnownSubclasses(name)) {
                 assignables.add(subclass.name());
             }
-            Collection<ClassInfo> implementors = beanDeployment.getIndex().getAllKnownImplementors(name);
-            for (ClassInfo implementor : implementors) {
+            for (ClassInfo implementor : beanDeployment.getBeanArchiveIndex().getAllKnownImplementors(name)) {
                 assignables.add(implementor.name());
             }
+            if (beanDeployment.hasApplicationIndex()) {
+                for (ClassInfo subclass : beanDeployment.getApplicationIndex().getAllKnownSubclasses(name)) {
+                    assignables.add(subclass.name());
```

Comment:
@mkouba `ClassInfo` doesn't override `equals`/`hashCode`(as noted in https://github.com/wildfly/jandex/issues/83), can you be sure you won't get duplicates from both indexes?

Reply:
That's OK. `assignables` is a set of dot names.

Reply:
Oops my bad, I thought its set of `ClassInfo`.